dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -606,10 +606,25 @@ class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink
       , heartbeat_recvd_count_(0)
       , hb_frag_recvd_count_(0)
       , participant_flags_(participant_flags)
+      , acknack_count_(0)
     { }
 
     bool should_nack() const;
     bool sends_directed_hb() const;
+    SequenceNumber preemptive_acknack_base() const
+    {
+      // RTI expects 0 while the spec implies it should be 1.
+      return (id_.guidPrefix[0] == 0x01 && id_.guidPrefix[1] == 0x01) ? 0 : 1;
+    }
+
+    CORBA::Long next_acknack_count()
+    {
+      // Reflect the heartbeat count for OpenDDS.
+      return (id_.guidPrefix[0] == 0x01 && id_.guidPrefix[1] == 0x03) ? heartbeat_recvd_count_ : ++acknack_count_;
```

Comment:
Could use the constant?

https://github.com/OpenDDS/OpenDDS/blob/e80a522c7ae1b60d0bec6a3691f046c86eb89d04/dds/DCPS/GuidUtils.h#L29

Could also make a function in `GuidUtils.h` that checks if `GUID_t` has a `GuidVendorId_t`.